dataclasses.py

```diff
@@ -602,3 +603,72 @@ def _astuple_inner(obj, tuple_factory):
                           for k, v in obj.items())
     else:
         return deepcopy(obj)
+
+
+def replace(obj, **changes):
+    """Return a new object replacing specified fields with new values. This
+    is especially useful for frozen classes.  Example usage::
+
+      @dataclass(frozen=True)
+      class C:
+          x: int
+          y: int
+
+      c = C(1, 2)
+      c1 = replace(c, x=3)
+      assert c1.x == 3 and c1.y == 2
+      """
+
+    if not isdataclass(obj):
+        raise TypeError("replace() should be called on dataclass instances")
+
+    # Keep track of the non-init fields to specify.
+    non_init_fields = {}
+
+    # For each field:
+    # If it's an init field and it's not in 'changes', then get it's
```

Comment:
its

Reply:
wow, and I'm usually the grammar police! Thanks. I'll address all of these small issues after I merge and review the whole file.